x/posts/internal/keeper/keeper_reactions.go

```diff
@@ -0,0 +1,150 @@
+package keeper
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/desmos-labs/desmos/x/posts/internal/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// -------------
+// --- PostReactions
+// -------------
+
+// SavePostReaction allows to save the given reaction inside the store.
+// It assumes that the given reaction is valid.
+// If another reaction from the same user for the same post and with the same value exists, returns an expError.
+// nolint: interfacer
+func (k Keeper) SavePostReaction(ctx sdk.Context, postID types.PostID, reaction types.PostReaction) error {
+	store := ctx.KVStore(k.StoreKey)
+	key := types.PostReactionsStoreKey(postID)
+
+	// Get the existent reactions
+	var reactions types.PostReactions
+	k.Cdc.MustUnmarshalBinaryBare(store.Get(key), &reactions)
+
+	// Check for double reactions
+	if reactions.ContainsReactionFrom(reaction.Owner, reaction.Value) {
+		return fmt.Errorf("%s has already reacted with %s to the post with id %s",
+			reaction.Owner, reaction.Value, postID)
+	}
+
+	// Check if the reaction is a registered one
+	post, _ := k.GetPost(ctx, postID)
+	if _, exist := k.DoesReactionForShortCodeExist(ctx, reaction.Value, post.Subspace); !exist {
+		return fmt.Errorf("reaction with short code %s isn't registered yet and can't be used to react to the post with ID %s and sub %s, please register it before use",
+			reaction.Value, postID, post.Subspace)
+	}
```

Comment:
I think it's better to move this inside the `Handler` instead so that this method can be called from the `InitGenesis` without much problems